src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientFactory.cs

```diff
@@ -3,20 +3,32 @@
 //
 using Azure.Core;
 using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.ConfigurationClients;
 using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
 {
     internal class ConfigurationClientFactory : IConfigurationClientFactory
     {
-        public ConfigurationClient CreateConfigurationClient(string connectionString, ConfigurationClientOptions clientOptions)
+        public IConfigurationClient CreateConfigurationClient(string connectionString, ConfigurationClientOptions clientOptions)
         {
-            return new ConfigurationClient(connectionString, clientOptions); 
+            var configurationClient = new ConfigurationClient(connectionString, clientOptions);
+            var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, "Endpoint"));
+            return new FailOverSupportedConfigurationClient(new List<LocalConfigurationClient>() { new LocalConfigurationClient(endpoint, configurationClient) });
```

Comment:
Why do we need this if the client is created from the connection string? We don't support multiple endpoints via connection string.

Reply:
Discussed with @jimmyca15 and came to a conclusion that even for a connection_string based client we should be using `FailOverSupportedConfigurationClient`.